src/main/java/uk/gov/hmcts/ccd/auditlog/AuditInterceptor.java

```diff
@@ -59,24 +61,33 @@ private AuditContext populateHttpSemantics(AuditContext auditContext,
         AuditContext context = (auditContext != null) ? auditContext : new AuditContext();
         context.setHttpStatus(response.getStatus());
 
-        if (request.getMethod() != null && httpMethodList.contains(request.getMethod().toUpperCase())) {
-            context.setHttpMethod(request.getMethod());
+        String requestMethod = request.getMethod();
+        if (requestMethod != null && httpMethodList.contains(requestMethod.toUpperCase())) {
+            context.setHttpMethod(requestMethod);
         } else {
             context.setHttpMethod(BAD_VALUE_TOKEN);
             LOG.error("Error while validating Http Method with value {} as it did not meet validation criteria:{}",
-                request.getMethod(), httpMethodList);
+                encodeString(requestMethod), httpMethodList);
         }
 
         context.setRequestPath(request.getRequestURI());
 
-        if (request.getHeader(REQUEST_ID) == null || request.getHeader(REQUEST_ID).matches(REQUEST_ID_PATTERN)) {
-            context.setRequestId(request.getHeader(REQUEST_ID));
+        String requestId = request.getHeader(REQUEST_ID);
+        if (requestId == null || requestId.matches(REQUEST_ID_PATTERN)) {
+            context.setRequestId(requestId);
         } else {
             context.setRequestId(BAD_VALUE_TOKEN);
             LOG.error("Error while validating Request Id with value {} as it did not meet validation criteria:{}",
-                request.getHeader(REQUEST_ID), REQUEST_ID_PATTERN);
+                encodeString(requestId), REQUEST_ID_PATTERN);
         }
         return context;
     }
 
+    private String encodeString(String value) {
+        if (value == null) {
+            value = "";
+        }
+
+        return Base64.getEncoder().encodeToString(value.getBytes(StandardCharsets.UTF_8));
```

Comment:
could you just encode the first 100 chars of the original string, to avoid logging too large values that may be intending DoS effect?

Reply:
This has been implemented with the new commit but Fortify doesn't accept it. If you see the below scan, Fortify tried a value more than 100 character and failed at L:95
https://emea.fortify.com/Releases/65394/Issues

Reply:
Fortify scan has been passed successfully. 
https://emea.fortify.com/Redirect/Releases/65394